sync/src/sys/mutex.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Rust mutex implementation used in SGX environments
+//!
+//! Per the docs and discussions Rust mutexes are not re-entrant,
+//! <https://github.com/rust-lang/rust/issues/32260>.
+
+#![allow(dead_code)]
+
+use mc_sgx_tstdc::Mutex as SgxMutex;
+
+/// The mutex backend to use with the common Rust std lib Mutex interface
+pub(crate) struct Mutex {
+    inner: SgxMutex,
+}
+
+unsafe impl Send for Mutex {}
+unsafe impl Sync for Mutex {}
+
+impl Mutex {
```

Comment:
I'm on the fence with this implementation along with https://github.com/mobilecoinfoundation/sgx/pull/256.
Doing it this way keeps the lower level C/SGX API out of this crate, but the abstraction here seems questionable.

The future plan is to take most of https://github.com/rust-lang/rust/blob/master/library/std/src/sync/mutex.rs and have this module contain the SGX specific logic. This means that this module will need to exist it's a matter of deciding if it's this thin abstraction or if the `sgx_thread_mutex_*` calls are invoked directly from this module